### PR TITLE
Image preview component added

### PR DESCRIPTION
**Pull request by konarshankar07**
## Purpose of this pull request

This pull request (PR) will introduce the masonry grid component documentation which is developed in https://github.com/magento/adobe-stock-integration scope and now part of https://github.com/magento/magento2

## Affected DevDocs pages

-  New topic

## Links to Magento source code

- https://github.com/magento/magento2/pull/25464

## What's New
Added the [ImagePreview component](https://devdocs.magento.com/guides/v2.3/ui_comp_guide/components/ui-image-preview.html) topic to the _UI Components Guide_.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.3/ui_comp_guide/components/ui-image-preview.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+group: ui-components-guide
+title: ImagePreview Component
+contributor_name: Shankar Konar
+contributor_link: https://github.com/konarshankar07
+---
+
+The ImagePreview component implements a detailed view image column. User can preview next or previous image by clicking next or previous arrow
```
Comment (dobooth) [inline guides/v2.3/ui_comp_guide/components/ui-image-preview.md]: Thanks for the contribution. Can you please clarify this sentence? I don't have a good idea of what this component actually does.

File: guides/v2.3/ui_comp_guide/components/ui-image-preview.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+| displayedRecord          | Currently previewed image.                                                                                                                                                                                         |  Object        | {}                                     |
+| lastOpenedImage          | Previously opened image.                                                                                                                                                                                           |  Int           | null                                   |
+| statefull                | Defines a list of component properties whose values are automatically saved in the configured storage if they change. <code>key</code> is the property's name and the <code>value</code> defines whether its saved.|  Object        | {sorting: true, lastOpenedImage: true} |
```
[the remark this answers is not in the document]
Comment (keharper) [reply guides/v2.3/ui_comp_guide/components/ui-image-preview.md]: Don't add HTML code to a table unless you have no other choice, such as using `</br>` elements to force a new line.

File: guides/v2.3/ui_comp_guide/components/ui-image-preview.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+The ImagePreview component implements a detailed view image column. User can preview next or previous image by clicking next or previous arrow
+
+## Dependencies
+
+The ImagePreview component has dependencies on the following components:
+
+*  Column: `app/code/Magento/Ui/view/base/web/js/grid/columns/column.js`
+
+## Configuration options
+
```
Comment (keharper) [inline guides/v2.3/ui_comp_guide/components/ui-image-preview.md]: Global in this table: Add backticks to all Option names and Default Values, except null

File: guides/v2.3/ui_comp_guide/components/ui-image-preview.md (hunk 1 above)
Comment (keharper) [reply guides/v2.3/ui_comp_guide/components/ui-image-preview.md]: Can you include a screen capture of this component?

Comment (filmaj) [reply guides/v2.3/ui_comp_guide/components/ui-image-preview.md]: Perhaps we could expand this introductory sentence a bit more. Here's my stab at it:

```suggestion
The ImagePreview component implements an inline detailed preview of an image, which is typically used in conjunction with a grid of image search results. Users can expand a more detailed preview of an image in-line with other image search results and also have the ability to click on "next" or "previous" arrows to navigate to the next or previous image's detailed preview.
```

File: guides/v2.3/ui_comp_guide/components/ui-image-preview.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+contributor_link: https://github.com/konarshankar07
+---
+
+The ImagePreview component implements a detailed view image column. User can preview next or previous image by clicking next or previous arrow
+
+## Dependencies
+
+The ImagePreview component has dependencies on the following components:
+
+*  Column: `app/code/Magento/Ui/view/base/web/js/grid/columns/column.js`
```
Comment (filmaj) [inline guides/v2.3/ui_comp_guide/components/ui-image-preview.md]: Is there a relevant separate section of documentation we can link to here?

**Comment by keharper** [comment]
running tests

**Comment by keharper** [comment]
running tests

**Comment by keharper** [comment]
running tests

**Comment by keharper** [comment]
running tests

**Comment by keharper** [comment]
Thanks for your contribution, @konarshankar07.

**Pull request merged**